src/service/signup.service.ts

```diff
@@ -13,6 +14,14 @@ export default {
       email,
       uid,
     }: { username: string; email: string; uid: string } = userdata;
+
+    // Validate username using the isSafeInput function
+    if (!validation.isSafeInput(username)) {
+      throw new Error(
+        'Invalid input. Please check your input for malicious characters.',
```

Comment:
This is to explicit to send back to the user. "Please check your inputs" is enough. To explicit error messages (send to the user) can be dangerous.